content/docs/instrumenting/clientlib_guidelines.md

```diff
@@ -152,71 +152,71 @@ A histogram MUST have the following methods:
+Labels are one of the [most powerful aspects](/docs/practices/instrumentation/#use-labels) of Prometheus, but [easily abused](/docs/practices/instrumentation/#do-not-overuse-labels). Accordingly client libraries must be very careful in how labels are offered to users.
 
 Client libraries MUST NOT under any circumstances allow users to have different label names for the same metric for Gauge/Counter/Summary/Histogram or any other Collector offered by the library.
 
 If your client library does validation of metrics at collect time, it MAY also verify this for custom Collectors.
 
 While labels are powerful, the majority of metrics will not have labels. Accordingly the API should allow for labels but not dominate it.
 
-A client library MUST allow for optionally specifying a list of label names at Gauge/Counter/Summary/Histogram creation time. A client library SHOULD support any number of label names. A client library MUST validate that label names meet the requirements in [http://prometheus.io/docs/concepts/data_model/#metric-names-and-labels](http://prometheus.io/docs/concepts/data_model/#metric-names-and-labels)
+A client library MUST allow for optionally specifying a list of label names at Gauge/Counter/Summary/Histogram creation time. A client library SHOULD support any number of label names. A client library MUST validate that label names meet the [documented requirements](/docs/concepts/data_model/#metric-names-and-labels).
 
 The general way to provide access to labeled dimension of a metric is via a `labels()` method that takes either a list of the label values or a map from label name to label value and returns a "Child". The usual `.inc()`/`.dec()`/`.observe()` etc. methods can then be called on the Child.
 
 The Child returned by `labels()` SHOULD be cacheable by the user, to avoid having to look it up again - this matters in latency-critical code.
 
 Metrics with labels SHOULD support a `remove()` method with the same signature as `labels()` that will remove a Child from the metric no longer exporting it, and a `clear()` method that removes all Children from the metric. These invalidate caching of Children.
 
-There SHOULD be a way to initialize a given Child with the default value, usually just calling `labels()`. No-labels metrics MUST always be initialized. See [http://prometheus.io/docs/practices/instrumentation/#avoid-missing-metrics](http://prometheus.io/docs/practices/instrumentation/#avoid-missing-metrics)
+There SHOULD be a way to initialize a given Child with the default value, usually just calling `labels()`. Metrics without labels MUST always be initialized to avoid [problems with missing metrics](/docs/practices/instrumentation/#avoid-missing-metrics).
 
-## **Metric names**
+### Naming
```

Comment:
The original heading of "metric names" is correct here, as this section is just about metric names. Naming of other things such as the API is covered elsewhere